rt/rs/extensions/rx2/src/main/java/org/apache/cxf/jaxrs/rx2/client/FlowableRxInvokerImpl.java

```diff
@@ -148,33 +151,122 @@ public FlowableRxInvokerImpl(WebClient wc, ExecutorService ex) {
     @Override
     public <T> Flowable<T> method(String name, Entity<?> entity, Class<T> responseType) {
         if (sc == null) {
-            return Flowable.fromFuture(wc.async().method(name, entity, responseType));
+            return Flowable.create(new FlowableOnSubscribe<T>() {
```

Comment:
If these 2 objects are the same and the only difference is the `subscribeOn`+`observerOn` when `sc != null`, in my opinion it would look nicer to create the `Flowable` just once and then apply the logic on it.

Reply:
The same comment should also apply to the changes in `ObservableRxInvokerImpl`.

Reply:
Agree.  Thanks for the comments.  Done.